fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/HttpClient.kt

```diff
@@ -74,13 +67,37 @@ internal class HttpClient(
         return retrieveResponse(request, connection)
     }
 
+    /**
+     * When method is Method.PATCH, use reflection to set private field "method" on the HttpURLConnection instance
+     * For other methods, fall back to the public setter "setRequestMethod"
+     */
+    private fun HttpURLConnection.forceMethod(method: Method) {
+        when (method) {
+            Method.PATCH -> {
+                try {
+                    (this.javaClass.getDeclaredField("delegate").apply { this.isAccessible = true }.get(this) as HttpURLConnection?)?.forceMethod(method)
+                } catch (ex: NoSuchFieldException) {
+                    // ignore
+                }
+                (listOf(this.javaClass.superclass, this.javaClass)).forEach {
+                    try {
```

Comment:
Please wrap the entire thing in `try catch` and remove the double nested try catch.